core/src/main/java/io/kestra/core/services/AbstractFlowTriggerService.java

```diff
@@ -0,0 +1,158 @@
+package io.kestra.core.services;
+
+import io.kestra.core.models.conditions.types.MultipleCondition;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.triggers.AbstractTrigger;
+import io.kestra.core.models.triggers.multipleflows.MultipleConditionStorageInterface;
+import io.kestra.core.models.triggers.multipleflows.MultipleConditionWindow;
+import io.kestra.core.runners.RunContextFactory;
+import jakarta.inject.Inject;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class AbstractFlowTriggerService<S extends MultipleConditionStorageInterface> {
+    @Inject
+    protected Optional<S> multipleConditionStorage;
```

Comment:
It seems that all runner have a multiple condition storage so why is it optional?